### ISSUE TITLE
Parse error: syntax error

### ISSUE THREAD

**Issue by thazer** [jveldboom/facebook-gallery#25]
Parse error: syntax error, unexpected '[', expecting ')' in /home/nicopart/public_html/facebook-gallery-master/class.facebook-gallery.php on line 11

PHP version: 5.3.28
with latest facebook-gallery class available on github...

on different hosting provider:
[21-Jan-2017 21:44:38 Europe/Bucharest] PHP Fatal error:  Class 'Facebook\Facebook' not found in /home/badromro/dev/facebook-gallery-master/class.facebook-gallery.php on line 11

**Comment by jveldboom** [comment]
The syntax error is due to PHP's updated array format. `$array = [];` I'm pretty sure it was added in PHP 5.6+. Try updating line 11 in the class to:

```php
$this->fb = new \Facebook\Facebook(array(
   'app_id' => $config['app_id'],
   'app_secret' => $config['app_secret'],
   'default_graph_version' => 'v2.8'
));
```

As for the class note found, it might also be caused by you running on an older version of PHP. You might want to try importing each class individually with just a simple require. For example:

```php
require_once('./facebook-sdk-v5/Facebook.php`);
```

**Comment by thazer** [comment]
After the changes the errors simply multiplied, thus i was forced to switch from php 5.3 to 5.6  but the Facebook classes just were not present, even when requiring facebook-sdk-v5/autoload.php and each class script gave error to:

[22-Jan-2017 10:27:34 Europe/Bucharest] PHP Fatal error:  Class 'Facebook\FacebookApp' not found in ./facebook-gallery-master/facebook-sdk-v5/Facebook.php on line 141
[22-Jan-2017 10:30:51 Europe/Bucharest] PHP Fatal error:  Class 'Facebook\FacebookClient' not found in ./facebook-gallery-master/facebook-sdk-v5/Facebook.php on line 159
[22-Jan-2017 10:32:28 Europe/Bucharest] PHP Fatal error:  Class 'Facebook\HttpClients\FacebookCurlHttpClient' not found in ./facebook-gallery-master/facebook-sdk-v5/FacebookClient.php on line 127
[22-Jan-2017 10:33:19 Europe/Bucharest] PHP Fatal error:  Interface 'Facebook\HttpClients\FacebookHttpClientInterface' not found in ./facebook-gallery-master/facebook-sdk-v5/HttpClients/FacebookCurlHttpClient.php on line 35

i've tryed to require each class but it only led to more fatal errors, emm please advise.
require_once('./facebook-sdk-v5/autoload.php');
require_once('./facebook-sdk-v5/Facebook.php');
require_once('./facebook-sdk-v5/FacebookApp.php');
require_once('./facebook-sdk-v5/FacebookClient.php');
require_once('./facebook-sdk-v5/HttpClients/FacebookCurlHttpClient.php');
require_once('./facebook-sdk-v5/HttpClients/FacebookHttpClientInterface.php');


p.s. i'm currently running php 7.0.14 ....

**Comment by thazer** [comment]
got it running in another way... i've downloaded a fresh copy of facebook php SDK from: https://github.com/facebook/php-graph-sdk/tree/5.4
and in class.facebook-gallery.php i've only required: require_once('./facebook-sdk-v5b/autoload.php');
and now your facebook-gallery script works. Thank you John!

As for anywone who hits my issues, PHP 5.6+ is reqired and as for the fbsdk present in your script, it's on you to check :D

**Comment by jveldboom** [comment]
👍  Thank you for looking into this and posting back the fix. I'll update the readme to mention that PHP 5.6+ is required. Looks like I also need to update the Facebook SDK to 5.4.

Will mark this issue closed once the SDK is updated to v5.4.
